### PR TITLE
Fixes #13470: Remove the use of Thread sessions for configuration

### PR DESCRIPTION
**Pull request by ehelms**
With the move to using a single user to authenticate to Pulp, the
thread extensions that generated a Runcible configuration for each
thread for a given user are no longer needed. This would further
reduce special code Katello has to carry and allow more Foreman
tests to pass without having to skip them when Katello is enabled.

This introduces a server object class and instantiates it on
startup with the configuration in SETTINGS and the remote login
as configured during installation.

### CODE REVIEW

**Comment by ehelms** [comment]
I am re-examining running Foreman tests with Katello enabled (https://github.com/Katello/katello/pull/4534) and found a lot of test failures due to our use of Thread in this instance. While doing so, it occurred to me that this should no longer be needed with our change to single user authentication for Pulp.

File: app/models/katello/concerns/user_extensions.rb [hunk 1]
```diff
@@ -14,12 +12,16 @@ def self.remote_user
         end
 
         def self.cp_oauth_header
-          fail Errors::UserNotSet, "unauthenticated to call a backend engine" if Thread.current[:cp_oauth_header].nil?
-          Thread.current[:cp_oauth_header]
+          #fail Errors::UserNotSet, "unauthenticated to call a backend engine" if Thread.current[:cp_oauth_header].nil?
+          #Thread.current[:cp_oauth_header]
```
Comment (jlsherrill) [inline app/models/katello/concerns/user_extensions.rb]: want to just remove these commented out lines?

File: app/services/katello/pulp/server.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            :logger => ::Foreman::Logging.logger('katello/pulp_rest'),
+            :exception => true,
+            :info => true,
+            :debug => true
+          }
+        )
+      end
+    end
+  end
+end
```
Comment (jlsherrill) [inline app/services/katello/pulp/server.rb]: both https://github.com/Katello/katello/pull/5750 and https://github.com/Katello/katello/pull/5710 are also adding ways to fetch a runcible instance for a particular capsule, would we want to try to provide that here?

**Comment by ehelms** [comment]
Addressed the removal of that comment, letting tests re-run

**Comment by ehelms** [comment]
Updated to account for this change -- https://github.com/Katello/katello/pull/5741/files#diff-4c67e2ccb02b5cf746fd11e670020715R56

**Comment by jlsherrill** [comment]
APJ

**Comment by ehelms** [comment]
[test]

**Pull request merged**